Support diff_match_patch_python library

jcushman opened the pull request:
Per #34, this PR adds support for an alternate diffing library in `annotate()`. We now have three options (though probably want to converge on one):

* By default, `annotate()` uses built in Python diffing.
* If you `pip install diff_match_patch`, then `annotate(use_dmp=True)` uses [this pure-python DMP library](https://pypi.org/project/diff-match-patch/).
* If you instead `pip install diff_match_patch_python`, then `annotate(use_dmp=True)` uses [this C++ DMP library](https://pypi.org/project/diff_match_patch_python/). (Note for testing that you must `pip uninstall` the other one first -- they don't coexist.)

The speed difference is *very* significant: in my test on the text in #34, the first option took 17,000ms to call annotate(), the second option took 700ms, and the third option took 10ms. So I'd pretty much only consider the `diff_match_patch_python` performance to be acceptable.

There's an open question in #34 of whether DMP returns correct results, though I haven't been able to reproduce the error.

(FWIW there's also some [packaging weirdness](https://github.com/JoshData/diff_match_patch-python/issues/4) about `diff_match_patch_python` -- it changes the interface of Google's official library but uses the same name, so if you have one installed you can't use packages that rely on the other one.)

mlissner commented:
Kind of sounds like we should just go with the last one. The advantage of the first is no dependencies, but it sounds like it's so slow it's not worth keeping around. The advantage of the second is...no C++ (?), but I feel like there's no getting around that in a big project these days anyhow? 

I guess if @mattdahl can't reproduce his error in #34, maybe we just go with the third one and drop the others? It wouldn't change our output or APIs, right, so seems like a safe decision? (I also don't think many people are using eyecite yet in any case.)

jcushman commented:
Updated to use `diff_match_patch_python` by default, but kept the option around to use `difflib` just for debugging. Also added Matt's long case as a testcase.

mlissner commented:
Nice. I especially appreciate the move to using `+`, `-` and `=`. It made the whole thing make more sense to me.

mattdahl commented:
This is great, thank you!
